ipysankeywidget/sankey_widget.py

```diff
@@ -42,6 +42,23 @@ class SankeyWidget(widgets.DOMWidget):
     def __init__(self, **kwargs):
         """Constructor"""
 
+        # check for duplicates in links
+        values = []
+        for link in kwargs.get('links', []):
+            linksource = link['source']
+            linktarget = link['target']
+            if 'type' in link:
+                linktype = link['type']
+            else:
+                linktype = None
+            if 'color' in link:
+                linkcolor = link['color']
+            else:
+                linkcolor = None
+            values.append((linksource, linktarget, linktype, linkcolor))
+        if len(values) != len(set(values)):
+            raise ValueError("Links have duplicates")
```

Comment:
A more helpful error message might be nice -- maybe something like

> links contains duplicate entries: source, target and type must be unique (see `https://github.com/ricklupton/ipysankeywidget/issues/22`)

or without the URL if you think that's a bad idea.

Keeping track of the index of the duplicate entry and including it in the error message would also be nice, but not necessary!